File: SQLite/example/ch01/examples/python/function.py
#!/usr/bin/env python

import string
from pysqlite2 import dbapi2 as sqlite

def hello_newman(*args):
    if len(args) > 0:
        return 'Hello %s' % string.join(args, ', ')
    return 'Hello Jerry'

con = sqlite.connect(":memory:")
con.create_function("hello_newman", -1, hello_newman)
cur = con.cursor()

cur.execute("select hello_newman()")
print cur.fetchone()[0] 

cur.execute("select hello_newman('Elaine')")
print cur.fetchone()[0] 

cur.execute("select hello_newman('Elaine', 'Jerry')")
print cur.fetchone()[0] 

cur.execute("select hello_newman('Elaine', 'Jerry', 'George')")
print cur.fetchone()[0] 

File: SQLite/example/ch01/examples/perl/function.pl
#!/usr/bin/env perl

use DBI;

sub hello_newman {
    if (@_ == 0) {
        return "Hello Jerry";
    } 

    return "Hello " . join ", ", @_;
}

# Connect
my $dbh = DBI->connect("dbi:SQLite:dbname=seinfeld.db", "", "", { RaiseError => 1 });

# Register function
$dbh->func('hello_newman', -1, \&hello_newman, 'create_function');

# Call it
print $dbh->selectrow_arrayref("SELECT hello_newman()")->[0] . "\n";
print $dbh->selectrow_arrayref("SELECT hello_newman('Elaine', 'Jerry')")->[0] . "\n";
print $dbh->selectrow_arrayref("SELECT hello_newman('Elaine', 'Jerry', 'George')")->[0] . "\n";

$dbh->disconnect;
